lib/Parse/ParseExpr.cpp

```diff
@@ -399,16 +399,23 @@ ParserResult<Expr> Parser::parseExprSequenceElement(Diag<> message,
   SyntaxParsingContext ElementContext(SyntaxContext,
                                       SyntaxContextKind::Expr);
 
-  if (shouldParseExperimentalConcurrency() && Tok.isContextualKeyword("await")) {
-    SourceLoc awaitLoc = consumeToken();
-    ParserResult<Expr> sub =
-      parseExprSequenceElement(diag::expected_expr_after_await, isExprBasic);
-    if (!sub.hasCodeCompletion() && !sub.isNull()) {
-      ElementContext.setCreateSyntax(SyntaxKind::AwaitExpr);
-      sub = makeParserResult(new (Context) AwaitExpr(awaitLoc, sub.get()));
-    }
+  if (Tok.isContextualKeyword("await")) {
+    if (shouldParseExperimentalConcurrency()) {
+      SourceLoc awaitLoc = consumeToken();
+      ParserResult<Expr> sub =
+        parseExprSequenceElement(diag::expected_expr_after_await, isExprBasic);
+      if (!sub.hasCodeCompletion() && !sub.isNull()) {
+        ElementContext.setCreateSyntax(SyntaxKind::AwaitExpr);
+        sub = makeParserResult(new (Context) AwaitExpr(awaitLoc, sub.get()));
+      }
 
-    return sub;
+      return sub;
+    } else {
+      // warn that future versions of Swift will parse this token differently.
+      diagnose(Tok.getLoc(), diag::warn_await_keyword)
+        .fixItInsert(Tok.getLoc(), "`")
```

Comment:
Would a fix-it replace would be an option here? e.g. `.fixItReplace(Tok.getLoc(), "`" + Tok.getText().str() + "`");` 
The same way [parseDecl](https://github.com/apple/swift/blob/e1f35f676d2f52e5927c326ecf32b59eeaed3eb2/lib/Parse/ParseDecl.cpp#L4587) does?

Reply:
Good point: this would also give a nicer-looking fix-it message on the command line (currently, it just shows the insertion of the two backticks and it looks awkward without the word "await" in the middle). Thanks.